### PR TITLE
common/pool: fix pool abuse

### PR DESCRIPTION
**Pull request by 0xcb9ff9**
### Description

by default `ants.NewPool` exist expiry, if submit job wait too long time `defer` function never call, like this

https://github.com/dogechain-lab/dbsc/blob/96778f1fc3647e466ec966c16935400bf1fa408c/core/bloombits/matcher.go#L167-L170

### CODE REVIEW

File: p2p/dial.go [hunk 1]
```diff
@@ -460,10 +454,10 @@ func (d *dialScheduler) startDial(task *dialTask) {
 	hkey := string(task.dest.ID().Bytes())
 	d.history.add(hkey, d.clock.Now().Add(dialHistoryExpiration))
 	d.dialing[task.dest.ID()] = task
-	gopool.Submit(func() {
+	go func() {
 		task.run(d)
```
Comment (DarianShawn) [inline p2p/dial.go:458]: Use time out context here to limit un-endless dial?

File: rpc/handler.go [hunk 2]
```diff
@@ -220,12 +218,12 @@ func (h *handler) cancelServerSubscriptions(err error) {
 // startCallProc runs fn in a new goroutine and starts tracking it in the h.calls wait group.
 func (h *handler) startCallProc(fn func(*callProc)) {
 	h.callWG.Add(1)
-	gopool.Submit(func() {
+	go func() {
 		ctx, cancel := context.WithCancel(h.rootCtx)
 		defer h.callWG.Done()
 		defer cancel()
 		fn(&callProc{ctx: ctx})
```
Comment (DarianShawn) [inline rpc/handler.go:225]: The context never expires, which should be fixed on the near future.

File: common/gopool/pool.go [hunk 3]
```diff
@@ -9,7 +9,7 @@ import (
 
 var (
 	// Init a instance pool when importing ants.
-	defaultPool, _   = ants.NewPool(ants.DefaultAntsPoolSize, ants.WithExpiryDuration(10*time.Second))
+	defaultPool, _   = ants.NewPool(ants.DefaultAntsPoolSize, ants.WithExpiryDuration(time.Hour))
```
Comment (DarianShawn) [inline common/gopool/pool.go]: Why 1 hour here?

**Review by DarianShawn — COMMENTED** [review]
Thanks for the great job.
The upstream repo makes something terrible, which we need to fix it step by step. Every small step counts. 🙏

Though I have some questions here.

File: p2p/dial.go (hunk 1 above)
Comment (0xcb9ff9) [reply p2p/dial.go:458]: by default TCP dial exist [OS level timeout](https://pkg.go.dev/net#Dialer), maybe wait upstream merge ?
https://github.com/dogechain-lab/dbsc/blob/691902b3b07786521dbc0766c357277a3c0eb370/p2p/dial.go#L66-L68

File: rpc/handler.go (hunk 2 above)
Comment (0xcb9ff9) [reply rpc/handler.go:225]: upstream PR https://github.com/ethereum/go-ethereum/pull/25457 better handle RPC timeout, wait BSC merge v1.12.2 ?

File: p2p/dial.go (hunk 1 above)
Comment (DarianShawn) [reply p2p/dial.go:458]: Not necessary.
We'll leave it here if it does expire.

File: rpc/handler.go (hunk 2 above)
Comment (DarianShawn) [reply rpc/handler.go:225]: Agree.
But we could merge it on the other PR.

File: common/gopool/pool.go (hunk 3 above)
Comment (DarianShawn) [reply common/gopool/pool.go]: If you don't like the idea of expiration tasks in the global pool, remove the expiry option in the global pool, create a volatile pool with required options will be an acceptable solution.

Comment (0xcb9ff9) [reply common/gopool/pool.go]: use `ants` version not exist `No Expiry` option, need upgrade version to >=2.6.0, I'm try upgrade 2.8.0  [e12de3c](https://github.com/dogechain-lab/dbsc/pull/27/commits/e12de3cbd2f9a46889da3cd99741d29011078713)

**Review by DarianShawn — APPROVED** [review]
Excellent. 🙌🏻

**Pull request merged**